### PR TITLE
Lots of changes.

### PR DESCRIPTION
**Pull request by Yaazarai**
Changed a few things. Added support for non-parented light objects & having the light system be able to be handled on different views which is defined in the light_init( shaders? , view_id ) script.

### CODE REVIEW

File: prettylight.gmx/scripts/light_set_offset.gml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+///light_set_offset( xoffset, yoffset );
+
+/**
+ * Set initial properties for a light object
```
Comment (niksudan) [inline prettylight.gmx/scripts/light_set_offset.gml:4]: Wrong description!

File: prettylight.gmx/scripts/light_set_offset.gml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * 
+ * @author FatalSleep
+ * @author Nik
```
Comment (niksudan) [inline prettylight.gmx/scripts/light_set_offset.gml:7]: I had nothing to do with this stop sharing the credit

File: prettylight.gmx/scripts/lights_draw.gml [hunk 3]
```diff
@@ -13,9 +13,11 @@
 surface_set_target( lights );
 draw_clear_alpha( alc, ala );
 draw_set_blend_mode( bm_add );
-with ( objLight ) {
-    draw_sprite_ext( ls , li , -view_xview + x + xo , -view_yview + y + yo , xs , ys , rs , cs , as );
-}
+for( var i = 0; i < ds_list_size( light_list ); i ++ ) {
```
Comment (niksudan) [inline prettylight.gmx/scripts/lights_draw.gml:16]: Space after the `for`

**Comment by niksudan** [comment]
This is great. Covers functionality for #1 too

File: README.md [hunk 4]
```diff
@@ -43,21 +43,24 @@ prettylight comes with a prebuilt light controller and parent object, but you ca
 
 The lights are controlled with the light controller object. This should have a pretty low depth so it runs before other objects. The controller object needs to run 4 functions:
 
-- ``` lights_init() ``` - this sets up the lighting system, call this in a **creation or trigger event**
+- ``` lights_init( shaders? , view_id ) ``` - this sets up the lighting system, call this in a **creation or trigger event**. "shaders?" true/false, defines whether or not to apply shaders to the lighting system. "view_id" is the id 0-15 of whichever view you wish to enable lighting on.
+
```
Comment (niksudan) [inline README.md:47]: Extra space here that needs deleting

**Comment by niksudan** [comment]
I'll take care of the changes.

**Comment by Yaazarai** [comment]
Thank you.

**Pull request merged**